File: Courzelo_Back/src/main/java/tn/esprit/courzelo/entities/AcademicProgramEntities/Class.java
package tn.esprit.courzelo.entities.AcademicProgramEntities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import tn.esprit.courzelo.entities.SessionEntities.Session;
import tn.esprit.courzelo.entities.UserCorzelo.UserCourzelo;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "Class")
public class Class {
    @Id
    private  String id;
    @Indexed
    private String name;
    @DBRef
    private Level level;
    @DBRef
    private List<Session> Sessions;
    @DBRef
    private List<Session> sessions;
    @DBRef
    private List<UserCourzelo> students;
    @DBRef
    private List<UserCourzelo> teachers;
}

File: Courzelo_Front/Courzelo/src/app/Models/AcademicProgramEntities/Class.ts
import { Session } from "../SessionEntities/Session";
import { UserCourzelo } from "../UserCorzelo/UserCourzelo";
import { Level } from "./Level";

export class Class{
    id:number |undefined;
    name!:string;
    level!: Level | undefined; 
    Schedule?:Array<Session> | undefined;
    Students?:Array<UserCourzelo> |undefined;
    
}